tests/queries/0_stateless/02003_toDateTimeOrNull_support_integer.sql

```diff
@@ -1,20 +1,32 @@
-SELECT toDateTimeOrNull(1583851242, 'Asia/Shanghai') = toDateTime(1583851242, 'Asia/Shanghai');
-SELECT toDateTime(1583851242, 'Asia/Shanghai') = toDateTimeOrZero(1583851242, 'Asia/Shanghai');
-
 SELECT toDate(123) = toDateOrNull(123);
-
 SELECT toDate(123) = toDateOrNull(toInt8(123));
 SELECT toDate(123) = toDateOrZero(toInt16(123));
 SELECT toDate(123) = toDateOrZero(toInt32(123));
 SELECT toDate(123) = toDateOrNull(toInt64(123));
 
+SELECT toDateOrNull(toInt8(123));
+SELECT toDateOrZero(toInt16(123));
+SELECT toDateOrZero(toInt32(1230));
+SELECT toDateOrNull(toInt64(12300));
```

Comment:
Just in case, can you, please, also add to tests cases when number will be out of allowed bounds and we get zero and null - all possible cases like: `select toDateOrZero(-1), toDate32OrNull(-200000000), select toDateOrNull(-20000000), select toDate32OrZero(1), select toDateOrZero(1), select toDateOrZero(100000000000)` and also `select toDate32OrZero(-1)` will be good to add.

Reply:
The `toDate(-1)` will call `FunctionConvert` return `1970-01-01`, and `toDateOrZero(-1)` will call `FunctionConvertWithExceptionMode` return `2149-06-06`, are they correct?
Maybe `FunctionConvertWithExceptionMode` try to convert Integer type to Date type by `static_cast` directly, but `FunctionConvert` use `DateTimeTransformImpl` with function `ToDateTransform32Or64Signed` make Integer to be signed integer first.

Reply:
> and toDateOrZero(-1) will call FunctionConvertWithExceptionMode return 2149-06-06, are they correct?

I would expect toDateOrZero(-1)  to return 1970-01-01 and toDateOrNull(-1) to return Null.
Probably you can just check the boundaries before calling ConverImpl.

Reply:
Ok, I would try to fix it.